File: proj.android_studio/ee-x-admob/src/main/java/com/ee/admob/AdMobBannerHelper.java
package com.ee.admob;

import android.content.Context;
import android.graphics.Point;

import androidx.annotation.NonNull;

import com.google.android.gms.ads.AdSize;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

import static com.google.common.truth.Truth.assertThat;

class AdMobBannerHelper {
    private Map<Integer, Point> _sizes;

    @NonNull
    private static Point convertAdSizeToSize(Context context, AdSize adSize) {
        int width = adSize.getWidthInPixels(context);
        int height = adSize.getHeightInPixels(context);
        return new Point(width, height);
    }

    AdMobBannerHelper(Context context) {
        _sizes = new HashMap<>();
        for (int i = 0; i < 4; ++i) {
            AdSize adSize = getAdSize(i);
            Point size = convertAdSizeToSize(context, adSize);
            _sizes.put(i, size);
        }
    }

    @NonNull
    AdSize getAdSize(int index) {
        if (index == 0) {
            return AdSize.BANNER;
        }
        if (index == 1) {
            return AdSize.LARGE_BANNER;
        }
        if (index == 2) {
            return AdSize.SMART_BANNER;
        }
        if (index == 3) {
            return AdSize.MEDIUM_RECTANGLE;
        }
        assertThat(false).isTrue();
        return AdSize.INVALID;
    }

    @NonNull
    Point getSize(int index) {
        return Objects.requireNonNull(_sizes.get(index));
    }
}

File: proj.android_studio/ee-x-admob/src/main/java/com/ee/admob/AdMobBannerHelper.kt
package com.ee.admob

import android.content.Context
import android.graphics.Point
import androidx.annotation.AnyThread
import com.google.android.gms.ads.AdSize
import com.google.common.truth.Truth.assertThat

internal class AdMobBannerHelper(context: Context) {
    companion object {
        @AnyThread
        private fun convertAdSizeToSize(context: Context, adSize: AdSize): Point {
            val width = adSize.getWidthInPixels(context)
            val height = adSize.getHeightInPixels(context)
            return Point(width, height)
        }
    }

    private val _sizes: MutableMap<Int, Point> = HashMap()

    init {
        for (i in 0..3) {
            val adSize = getAdSize(i)
            val size = convertAdSizeToSize(context, adSize)
            _sizes[i] = size
        }
    }

    @AnyThread
    fun getAdSize(index: Int): AdSize {
        if (index == 0) {
            return AdSize.BANNER
        }
        if (index == 1) {
            return AdSize.LARGE_BANNER
        }
        if (index == 2) {
            return AdSize.SMART_BANNER
        }
        if (index == 3) {
            return AdSize.MEDIUM_RECTANGLE
        }
        assertThat(false).isTrue()
        return AdSize.INVALID
    }

    @AnyThread
    fun getSize(index: Int): Point {
        return _sizes[index] ?: throw IllegalArgumentException("Invalid size index")
    }
}